template/template.go

```diff
@@ -157,34 +181,37 @@ func (e *executer) resolveComplexMessage(message ast.ComplexMessage) error {
 	return resolutionErr
 }
 
-func (e *executer) resolveDeclarations(declarations []ast.Declaration) error {
-	m := make(map[ast.Value]struct{}, len(declarations))
+func newLiteral(v any) ast.Value { //nolint:ireturn
+	switch t := v.(type) {
+	case string:
+		return ast.QuotedLiteral(t)
+	case float64:
+		return ast.NumberLiteral(t)
+	case int:
+		return ast.NumberLiteral(t)
+	}
+
+	return nil
+}
 
+func (e *executer) resolveDeclarations(declarations []ast.Declaration) error {
 	for _, decl := range declarations {
 		switch d := decl.(type) {
 		case ast.ReservedStatement:
 			return fmt.Errorf("%w", mf2.ErrUnsupportedStatement)
 		case ast.LocalDeclaration:
-			m[d.Variable] = struct{}{}
-
-			resolved, err := e.resolveExpression(d.Expression)
-			// if can't resolve the expression, leave it as unresolved, e.g. {$foo}
-			e.variables[string(d.Variable)] = resolved
-
-			if err != nil {
-				return fmt.Errorf("local declaration: %w", err)
-			}
-
+			e.variables[string(d.Variable)] = newVariable(d.Expression, nil)
 		case ast.InputDeclaration:
-			m[d.Operand] = struct{}{}
+			name := string(d.Operand.(ast.Variable)) //nolint: forcetypeassert // always ast.Variable
```

Comment:
Can we use `ast.Variable` as key T in the map? To avoid converting `ast.Variable` to `string` all the time.

Reply:
Template params are saved in that map as `string: any`, this change would require converting strings into `ast.Variable`